### PR TITLE
AppArmor loader DaemonSet

### PR DESCRIPTION
**Pull request by timstclair**
This PR adds a proof-of-concept DaemonSet for loading AppArmor profiles from a ConfigMap onto a cluster.

The DaemonSet runs a small go program that is mostly a wrapper around `apparmor_parser`, which actually loads the profiles. The additional features the loader provides are:
- Polling
- Sourcing from multiple directories
- Better handling of symlinks (necessary for ConfigMap volumes)
- Better logging and messaging

This is meant to provide an example of using AppArmor, and be a starting point for users trying out AppArmor or building there own solutions.

/cc @jfrazelle @thockin

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/contrib/1658)

### CODE REVIEW

File: apparmor/loader/example-configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+# An example ConfigMap demonstrating how profiles can be stored as Kubernetes objects, and loaded by
+# the DaemonSet.
```
Comment (dchen1107) [inline apparmor/loader/example-configmap.yaml]: s/DaemonSet/AppArmor Profile Loader DaemonSet?

Comment (timstclair) [inline apparmor/loader/example-configmap.yaml]: Done. I went with "apparmor-loader DaemonSet" since that's the actual object name.

**Comment by dchen1107** [comment]
LGTM to me overall. But not sure why travis complained the following:

```
This job is running on container-based infrastructure, which does not allow use of 'sudo', setuid and setguid executables.
If you require sudo, add 'sudo: required' to your .travis.yml
```

**Comment by timstclair** [comment]
I think the failure was just a flake. It shows the sudo message for every build.

**Comment by timstclair** [comment]
Squashed commits.

**Comment by dchen1107** [comment]
LGTM

**Pull request merged**